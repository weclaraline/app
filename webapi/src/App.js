const fileReader = require("./lib/utils/FileReader");
const xmlParser = require("./lib/utils/XMLParser");
const InvoiceAnalyzer = require("./lib/analysis/InvoiceAnalyzer");
const {
  getFormaPago,
  getVersion,
  getUsoCFDI,
} = require("./lib/InvoiceFieldResolver");

// fileReader.readFile("../XML/sgm.xml", function (data) {
//   let parsed = xmlParser.ParseXMLString(data);
//   console.log(parsed);
// });

async function app() {
  let data = await fileReader.readFile("../XML/hmotrosconceptos.xml");
  let parsed = xmlParser.ParseXMLString(data);

  console.log(parsed);

  console.log(getVersion(parsed));

  console.log(getFormaPago(parsed));
  console.log(getUsoCFDI(parsed));
  let analyzer = new InvoiceAnalyzer();

  const res = analyzer.analyze(parsed);
  console.log(res)
}

app();
