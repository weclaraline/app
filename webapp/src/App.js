import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import 'fontsource-roboto';

import './App.css';
import Login from './components/login/Login';
import Logout from './components/logout/Logout';
import Navbar from './components/navbar/Navbar';
//import HomePage from './pages/HomePage';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Navbar />
        <Login></Login>
        <Logout></Logout>
      </BrowserRouter>
    </div>
  );
}

export default App;
