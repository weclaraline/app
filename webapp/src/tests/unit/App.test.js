import { render, unmountComponentAtNode } from "react-dom";
import { getByTestId, waitFor } from "@testing-library/react";
import localForage from "localforage";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import "./testing_utils";
import App from "../../App";

const build = () => {render(<App />, container);}

let container = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});
  
afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
    return container;
});

it("Renders original content", ()=> {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders user login when user is not logged in", async () => {
    build();
    await waitFor( () => { 
        expect(getByTestId(container, "login_screen")).toBeInTheDocument() 
    });
});

it("renders user HomePage when user is logged in",async()=> {
    const CURRENT_LOGGED_USER = "user";
    await localForage.setItem(CURRENT_LOGGED_USER, {notEmtpyObject:{}});
    build();
    await waitFor(() => {
        expect(getByTestId(container, "home_page")).toBeInTheDocument() 
    });
});