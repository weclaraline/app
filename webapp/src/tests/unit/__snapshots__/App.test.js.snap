// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<div
  className="App"
>
  <ThemeProvider>
    <BrowserRouter>
      <Route
        path="/"
      >
        <Login />
      </Route>
    </BrowserRouter>
  </ThemeProvider>
</div>
`;
